### DISCUSSION TITLE
Possible Error with AutoGen Services?

### DISCUSSION THREAD

**Discussion by Ryan-Britton** [ServiceStack/Discuss#117] [Q&A] [answered]
Hi,

I'm trying to generate c# classes and autoquery services following the tutorial [here](https://docs.servicestack.net/autoquery/autogen#code-generation-of-autoquery-crud-services)

after creating a new project with "x new empty AutoGenTest" and running "x mix autoquery postgres" and correctly configuring the DbConnectionFactory, and adding the following : 

`
Plugins.Add(new AutoQueryFeature {
    MaxLimit = 1000,
    GenerateCrudServices = new GenerateCrudServices {}
});
`

I run into a problem where both DbFactory and ServiceStackHost.Instance are null on line 767 of GenerateCrudServices.cs : 

`
 var dbFactory = DbFactory ?? ServiceStackHost.Instance?.TryResolve<IDbConnectionFactory>()
            ?? throw new NotSupportedException(nameof(GenerateCrudServices) + ".DbFactory is not configured");
`

I can overcome this by manually setting the DbFactory property on the GenerateCrudServices instance - and that allows me to correctly generate the AutoQuery services from the database (which I can see in the API Explorer by setting AutoRegister = true).

however - when I try to generate the code (either by running 'x csharp https://localhost:5001 -path /crud/all/csharp' from the command line or visiting 'https://localhost:5001/crud/all/csharp') I get a 405 : method not allowed error : 

![image](https://github.com/ServiceStack/Discuss/assets/124330161/a1069ac2-d678-4fd2-a259-d5579c175b37)

Is this something you guys are aware of, or can suggest a workaround for?

**Comment by mythz** [comment, marked as answer]
Sounds like you're using the newer templates with Endpoint Routing, which was missing the `IGet` Verb marker to register it as a GET Endpoint route. Can you try with the latest v8.2.3 [Pre Release](https://docs.servicestack.net/pre-release) packages?

**Reply by Ryan-Britton** [reply]
that worked - thank you 😊
